[ML] Data Frame Analytics results explorer: styling update

alvarezmelissa87 opened the pull request:
## Summary

Fixes https://github.com/elastic/kibana/issues/146586

This PR updates the DFA results view layout and style. Updates the functional tests. Flaky test runner: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/1977

Before: 

<img width="1184" alt="image" src="https://user-images.githubusercontent.com/6446462/220760702-a6259a26-b6c6-47c3-8f70-311283e2ad40.png">

After:

<img width="979" alt="image" src="https://user-images.githubusercontent.com/6446462/222531691-5af5ac34-4d3c-4a17-819e-1eb02bd7c0fe.png">

<img width="963" alt="image" src="https://user-images.githubusercontent.com/6446462/222531769-69f4ec18-63b0-4c1b-aed5-6289fa4c6d1e.png">


### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

peteharverson commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/expandable_section/expandable_section.tsx:
```diff
@@ -82,54 +82,58 @@ export const ExpandableSection: FC<ExpandableSectionProps> = ({
+                  data-test-subj={`mlDFExpandableSection-${dataTestId}-toggle-button`}
+                >
+                  <EuiText size="s" color="default">
+                    <h5>{title}</h5>
```
Is there anything we can do with the styling of the tab title? I feel it gets sort of lost as the font size / styling is very similar to the rest of the tab header content. And does it need to be `h5`? I suspect that breaks accessibility rules about levels of `h` elements if the page doesn't have `h1` to `h4` already.

<img width="503" alt="image" src="https://user-images.githubusercontent.com/7405507/221222689-af223ae4-d19f-46f1-ab9d-2d74e454c614.png">

peteharverson replied:
Latest styling edits look good here!

alvarezmelissa87 commented:
@elasticmachine merge upstream

mdefazio reviewed (changes requested):
A few polishing ideas:

Update the padding around expandable sections so it's symmetrical. Would suggest `8px 16px`.  This helps the arrow sit evenly within the row.
Before:
<img width="449" alt="image" src="https://user-images.githubusercontent.com/3756330/222189511-13325a73-e79a-4f4f-9dd9-68c9173c1cbf.png">

padding: 8px
<img width="797" alt="image" src="https://user-images.githubusercontent.com/3756330/222190415-52b6ea84-cb2e-4f68-b79b-f9020573be4c.png">

padding-left/right: 16px
<img width="825" alt="image" src="https://user-images.githubusercontent.com/3756330/222190565-2dea60dc-4867-4bfb-ab19-ad801a97bed9.png">

- - -

I agree with @peteharverson regarding the heading of the section getting lost a bit.  Here's an update to the design to help make it more visible (hopefully :D). 

* Changes 'Type', 'Source index' and 'Destination index' to just be text, rather than badges.  Should help make those feel so dominant
* Increases font size of the the row heading.  This should be larger than the headings within the expanded row.  
* I've also removed the border above the tabs to help reduce the extra visual elements.  I think white-space is enough to clear define these sections.

Happy to chat about this as well and find some easier wins based on any technical limitations.

![image](https://user-images.githubusercontent.com/3756330/222201357-77572ef6-8001-42aa-bd0b-42581da8de15.png)

mdefazio commented:
As a side comment, how valuable is this Phases block?  It's currently in a prominent location and probably something the user just wants to pass over anyway unless there were issues—is that correct?  I'd just want to know this completed all phases and if there are any errors/warnings, but this level of detail could probably be removed here or hidden behind a click.

![image](https://user-images.githubusercontent.com/3756330/222204114-bd0f585f-46da-4c7a-b764-457885c30c11.png)

alvarezmelissa87 commented:
@peteharverson, @mdefazio - I've made a few updates based on your suggestions (updated in 47b8b20972131ca8eb6f2db3859f7c1dd247b47f and updated screenshots in the description)
- increased tab title font
- switched badge back to the value 
   - @mdefazio - went with keeping the badges to stay consistent with the other result sections
- removed the horizontal line between the tab title and the sub tabs
- added the padding to align the tab header items

alvarezmelissa87 commented:
@elasticmachine merge upstream

mdefazio reviewed (changes requested):
Thanks for making those edits!  A few things i'm noticing:

Looks like the order got reversed for these or am I imagining this?
<img width="803" alt="image" src="https://user-images.githubusercontent.com/3756330/222464303-dc69627a-210b-46b7-9c12-82d5601bb4e2.png">

And I think the expanded row content could use a bit more padding around it: 
`.mlExpandedRowDetails { padding: 8px 16px 16px; }`

Guessing the 'Failed' badge should be a different color? Our danger color perhaps?  Apologies for not accounting for this in my mockup. Are the statuses [Started, Stopped, Failed]?

Also noticing here the phases when something does fail.  Certainly makes sense here and we can come back to this in future versions on how this might change if everything has passed.  One note on it though: Wouldn't it make sense for it to read **`Phases: 2 of 8 complete`**. Rather than just showing the current phase? 

<img width="1194" alt="image" src="https://user-images.githubusercontent.com/3756330/222475846-c1925ac0-eaa7-4661-8057-3c33a690ba28.png">

mdefazio reviewed (approved):
LGTM! Thank you!

peteharverson reviewed (approved):
Tested latest styling edits (Chrome and Firefox) and LGTM

alvarezmelissa87 commented:
@elasticmachine merge upstream
